### PR TITLE
Feature: Header / Dropdown / Language / i18n

### PR DESCRIPTION
**Pull request by moiskillnadne**
https://mindlogger.atlassian.net/browse/ML-131

Features list:
- Header Component (Includes Logo, language dropdown, login button | Excludes User profile dropdown)
- Routes
- i18n types
- useCustomTranslation
- SHARED: Dropdown
- SHARED: NavbarButton

### CODE REVIEW

**Review by BamMironov — CHANGES_REQUESTED** [review]

File: src/app/layout.tsx [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
   return (
     <>
-      <div>Translations: {t("Landing.title")}</div>
+      <Header />
       <div>{props.children}</div>
-      <div>Some footer</div>
+      <div>Some footer: location - {location.pathname}</div>
     </>
   )
 }
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/app/layout.tsx:20]: Done

File: src/app/providers/route.tsx [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+interface RouteProviderProps {
+  children: ReactNode
+}
+
+const RouteProvider = ({ children }: RouteProviderProps) => {
+  return <BrowserRouter>{children}</BrowserRouter>
+}
+
+export default RouteProvider
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/app/providers/route.tsx:12]: Done

File: src/features/language/lib/useLanguageList.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+export const useLanguageList = (): UseLanguageListOutput => {
+  const preparedLanguageList = enumToArray(SupportableLanguage).map(lang => ({
+    localizationPath: lang.key.toLocaleLowerCase(), // Enum keys should be equal list of keys in localizations file, because enum key = localization path
+    eventKey: lang.value,
+  }))
+
+  return {
+    preparedLanguageList,
+  }
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/features/language/lib/useLanguageList.ts:23]: Done

File: src/features/language/ui/LanguageDropdown.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+export interface LanguageDropdownProps {
+  onSelectExtended?: () => void
+}
+
+const LanguageDropdown = ({ onSelectExtended }: LanguageDropdownProps) => {
+  const { t, i18n } = useLanguageTranslation()
+  const [language, setLanguage] = useState(i18n.language || SupportableLanguage.English)
+  const { preparedLanguageList } = useLanguageList()
+
+  const onSelectHandler = (lang: string | null) => {
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/features/language/ui/LanguageDropdown.tsx:18]: Done

File: src/shared/Dropdown/lib/interfaces.ts [hunk 5]
```diff
@@ -0,0 +1,6 @@
+export interface DropdownOption {
+  eventKey: string
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/shared/Dropdown/lib/interfaces.ts:2]: Done

File: src/widgets/header/ui/index.tsx [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  const closeExpandedNavbar = () => {
+    setExpanded(false)
+  }
+
+  const onLogoClickHandler = () => {
+    navigate(ROUTES.main.path)
+    closeExpandedNavbar()
+  }
+
+  const onNavbarToggleHandler = () => {
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/widgets/header/ui/index.tsx:29]: Done

File: src/widgets/header/ui/index.tsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+  const { t } = useNavbarTranslation()
+  const navigate = useNavigate()
+
+  const [expanded, setExpanded] = useState<boolean>(false)
+
+  const closeExpandedNavbar = () => {
+    setExpanded(false)
+  }
+
+  const onLogoClickHandler = () => {
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/widgets/header/ui/index.tsx:24]: Done

File: src/features/language/lib/useLanguageList.ts [hunk 8]
```diff
@@ -0,0 +1,24 @@
+import { SupportableLanguage } from "../../../app/system/locale/constants"
+import { enumToArray } from "../../../utils/enumToArray"
```
[the remark this answers is not in the document]
Comment (moiskillnadne) [reply src/features/language/lib/useLanguageList.ts]: Done

**Review by BamMironov — APPROVED** [review]

**Pull request merged**